examples/server/server.cpp

```diff
@@ -405,127 +473,620 @@ bool parse_str_to_bool(const std::string & s) {
+    if (!tinydiarize_str.empty()) params.tinydiarize = parse_str_to_bool(tinydiarize_str);
+    std::string split_on_word_str = get_param_value("split_on_word");
+    if (!split_on_word_str.empty()) params.split_on_word = parse_str_to_bool(split_on_word_str);
+    std::string no_timestamps_str = get_param_value("no_timestamps");
+    if (!no_timestamps_str.empty()) params.no_timestamps = parse_str_to_bool(no_timestamps_str);
+    std::string language_str = get_param_value("language");
+    if (!language_str.empty()) params.language = language_str;
+    std::string detect_language_str = get_param_value("detect_language");
+    if (!detect_language_str.empty()) params.detect_language = parse_str_to_bool(detect_language_str);
+    std::string prompt_str = get_param_value("prompt");
+    if (!prompt_str.empty()) params.prompt = prompt_str;
+    std::string response_format_str = get_param_value("response_format");
+    if (!response_format_str.empty()) params.response_format = response_format_str;
+    std::string suppress_non_speech_str = get_param_value("suppress_non_speech");
+    if (!suppress_non_speech_str.empty()) params.suppress_nst = parse_str_to_bool(suppress_non_speech_str);
+    std::string suppress_nst_str = get_param_value("suppress_nst");
+    if (!suppress_nst_str.empty()) params.suppress_nst = parse_str_to_bool(suppress_nst_str);
+    std::string no_context_str = get_param_value("no_context");
+    if (!no_context_str.empty()) params.no_context = parse_str_to_bool(no_context_str);
+    return true; // Indicate success
+}
+
+// --- Task Queue and Worker Thread for Concurrency ---
+struct TranscriptionTask {
+    int id;
+    std::shared_ptr<std::vector<float>> audio_data_pcmf32_ptr;
+    std::shared_ptr<std::vector<std::vector<float>>> audio_data_pcmf32s_ptr;
+    whisper_params params;
+    std::promise<json> result_promise;
+    std::string original_filename;
+    std::shared_ptr<std::atomic<bool>> connection_alive;
+    std::shared_ptr<std::atomic<bool>> cancel_flag;
+};
+
+std::queue<TranscriptionTask> task_queue;
+std::mutex task_queue_mutex;
+std::condition_variable task_queue_cv;
+std::atomic<int> next_task_id{0};
+
+// RAII for whisper_context
+using ContextPtr = std::unique_ptr<whisper_context, decltype(&whisper_free)>;
+
+std::vector<ContextPtr> context_pool;
+std::vector<std::thread> worker_threads;
+std::mutex contextPoolMutex;
+std::mutex contextManagementMutex;
+
+// RAII thread guard for monitor thread
+class ThreadGuard { // Renamed
+    std::thread t;
+public:
+    explicit ThreadGuard(std::thread&& thr) : t(std::move(thr)) {}
+    ~ThreadGuard() { if (t.joinable()) t.join(); }
+    ThreadGuard(const ThreadGuard&) = delete;
+    ThreadGuard& operator=(const ThreadGuard&) = delete;
+    ThreadGuard(ThreadGuard&& other) noexcept : t(std::move(other.t)) {}
+    ThreadGuard& operator=(ThreadGuard&& other) noexcept { if (t.joinable()) t.join(); t = std::move(other.t); return *this; }
```

Comment:
Nit: implementation of function could be on a separate/new lines for readability.